paasta_tools/vitesscluster_tools.py

```diff
@@ -1,31 +1,581 @@
+    config = VtAdminConfigDict(
+        cells=cells,
+        apiAddresses=["http://localhost:15000"],
+        affinity={"nodeAffinity": node_affinity},
+        extraLabels=labels,
+        extraFlags=VTADMIN_EXTRA_FLAGS,
+        extraEnv=env,
+        replicas=replicas,
+        readOnly=False,
+        apiResources={
+            "requests": requests,
+            "limits": requests,
+        },
+        webResources={
+            "requests": requests,
+            "limits": requests,
+        },
+    )
+    return config
+
+
+def get_tablet_pool_config(
+    cell: str,
+    db_name: str,
+    keyspace: str,
+    port: str,
+    zk_address: str,
+    throttle_query_table: str,
+    throttle_metrics_threshold: str,
+    tablet_type: str,
+    region: str,
+    vttablet_resources: Dict[str, str],
+    env: List[Dict[str, Any]],
+    labels: Dict[str, str],
+    node_affinity: dict,
+) -> TabletPoolDict:
+    """
+    get vttablet config
+    """
+    vttablet_extra_flags = VTTABLET_EXTRA_FLAGS.copy()
+    flag_overrides = {
+        "throttle_metrics_query": f"select max_replication_delay from max_mysql_replication_delay.{throttle_query_table};",
+        "throttle_metrics_threshold": throttle_metrics_threshold,
+        "enforce-tableacl-config": "true",
+        "table-acl-config": f"/etc/vitess_keyspace_acls/acls_for_{db_name}.json",
+        "table-acl-config-reload-interval": "60s",
+        "queryserver-config-strict-table-acl": "true",
+        "db-credentials-server": "vault",
+        "db-credentials-vault-addr": f"https://vault-dre.{region}.yelpcorp.com:8200",
+        "db-credentials-vault-path": "secrets/vitess/vt-tablet/vttablet_credentials.json",
+        "db-credentials-vault-tls-ca": f"/etc/vault/all_cas/acm-privateca-{region}.crt",
+        "db-credentials-vault-ttl": "60s",
+    }
+    vttablet_extra_flags.update(flag_overrides)
+
+    environment_overrides: Dict[str, Any] = {
+        "VAULT_ADDR": f"https://vault-dre.{region}.yelpcorp.com:8200",
+        "VAULT_CACERT": f"/etc/vault/all_cas/acm-privateca-{region}.crt",
+        "TOPOLOGY_FLAGS": f"--topo_implementation {TOPO_IMPLEMENTATION} --topo_global_server_address ${zk_address} --topo_global_root {TOPO_GLOBAL_ROOT}",
+        "CELL_TOPOLOGY_SERVERS": zk_address,
+        "DB": db_name,
+        "KEYSPACE": keyspace,
+    }
+    environment_overrides.update(VTTABLET_EXTRA_ENV)
+    updated_vttablet_extra_env = (
+        get_updated_environment_variables(environment_overrides) + env
+    )
+
+    # Add extra pod label to filter
+    tablet_type_label = limit_size_with_hash(name=f"{db_name}_{tablet_type}", limit=63)
+    labels.update({"tablet_type": tablet_type_label})
+
+    try:
+        type = load_system_paasta_config().get_vitess_tablet_pool_type_mapping()[
+            tablet_type
+        ]
+    except KeyError:
+        log.error(
+            f"Tablet type {tablet_type} not found in system paasta config vitess_tablet_pool_type_mapping"
+        )
```

Comment:
should we exit early/default to another type if we run into this case? or does this `type` not being found not prevent us from creating usable pods?

Reply:
This attaches a type of vttablet created, which is either externalmaster or externalreplica in our case. We can definitely make this an exception with a default value

Reply:
Fixed in https://github.com/Yelp/paasta/pull/3889/commits/2f6ecc76417760dbd1cb424c3b61dabd5c83ea09